homeassistant/components/google/api.py

```diff
@@ -0,0 +1,179 @@
+import logging
+from typing import Dict, List
+
+from google.oauth2.credentials import Credentials
+from googleapiclient.discovery import build
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_CLIENT_ID, CONF_CLIENT_SECRET
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.config_entry_oauth2_flow import (
+    AbstractOAuth2Implementation,
+    OAuth2Session,
+)
+
+from .const import GOOGLE_CALENDAR_API, GOOGLE_PEOPLE_API, SCOPES
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class GoogleAPI(ABC):
+    """
+    Google APIs & Services.
+
+    https://console.developers.google.com/apis/dashboard
+    """
+
+    def __init__(self, client_id: str, client_secret: str, info: Dict):
+        """Initialize Google API."""
+        info["client_id"] = client_id
+        info["client_secret"] = client_secret
+        scopes = info.get("scope", "").split(" ")
+        self.creds = Credentials.from_authorized_user_info(info, scopes)
+        self.calendar = None
+        self.people = None
+        self.info = info
+
+    def setup(self) -> bool:
+        """Set up Google APIs and Services."""
+
+        if self.creds.has_scopes(SCOPES[GOOGLE_CALENDAR_API]):
+            self.calendar = GoogleCalendarAPI(self.creds)
+
+        if self.creds.has_scopes(SCOPES[GOOGLE_PEOPLE_API]):
+            self.people = GooglePeopleAPI(self.creds)
+
+        return True
+
+
+class GoogleCalendarAPI:
+    """
+    Google Calendar API.
+
+    https://developers.google.com/calendar/auth
+    https://developers.google.com/calendar
+    https://developers.google.com/resources/api-libraries/documentation/calendar/v3/python/latest/index.html
+    https://developers.google.com/calendar/concepts/events-calendars#calendar_and_calendar_list
+    TODO: Push updates # pylint: disable=fixme
+    https://developers.google.com/calendar/v3/push
+    """
+
+    def __init__(self, creds: Credentials):
+        """Initialize Google Calendar API."""
+        self._api = build("calendar", "v3", credentials=creds, cache_discovery=False)
+
+    def list_calendars(self) -> List:
+        """
+        List calendars.
+
+        https://developers.google.com/calendar/v3/reference/calendarList/list
+        """
+        calendars = (
+            self._api.calendarList().list().execute()  # pylint: disable=maybe-no-member
+        )
+        return calendars.get("items", [])
+
+    def list_events(self, calendar_id="primary", count=None) -> List:
+        """
+        List events.
+
+        https://developers.google.com/calendar/v3/reference/events/list
+        """
+        # TODO: List events for this calendar month # pylint: disable=fixme
+        # TODO: List events for the next 7 days # pylint: disable=fixme
+        # TODO: List events for today # pylint: disable=fixme
+        # TODO: List x scheduled/upcoming events # pylint: disable=fixme
+        date_start = (
+            datetime.datetime.utcnow().isoformat() + "Z"
```

Comment:
Please use `utcnow` imported from `homeassistant.util.dt`. It will have the `Z` in isoformat.